lib/config/config-ops.js

```diff
@@ -268,5 +272,84 @@ module.exports = {
      */
     isEverySeverityValid(config) {
         return Object.keys(config).every(ruleId => this.isValidSeverity(config[ruleId]));
+    },
+
+    /**
+     * Merges all configurations in a given config vector. A vector is an array of config file paths, each optionally
+     * followed by one or more numbers that correspond to the indices of nested config blocks within the config file's
+     * overrides section. All config data is assumed to be cached.
+     * @param {Array<number|string>} vector array of config file paths or relative override indices
+     * @returns {Object} config object
+     */
+    getConfigFromVector(vector) {
+
+        const subvector = Array.from(vector);
+        let config,
```

Comment:
The `config` variable is reassigned in several different places, and it's mutated later on, which makes this code hard to follow. Could it be restructured into functions to make it easier to tell what's going on?